### PR TITLE
added poll and vote commands

### PR DESCRIPTION
**Pull request by SizzlinSeal01**
i tested it on a tester bot and it worked

### CODE REVIEW

File: commands/polls/polls.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            memberName: 'poll',
+            description: 'Creates a poll with up to 10 choices.',
+            examples: ["!poll \"What\'s your favourite food?\" \"Hot Dogs,Pizza,Burgers,Fruits,Veggies\" 10"],
+            args: [
+                {
+                    key: 'question',
+                    prompt: 'What is the poll question?',
+                    type: 'string',
+                    validate: question => {
+                        if (question.length < 101 && question.length > 11) return true;
```
Comment (MMJ744) [inline commands/polls/polls.js]: Is there a reason for this limit? I understand why an upper limit of some form could be needed but I see nothing wrong with a one word poll question.

File: commands/polls/polls.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                        return 'Polling options must be greater than one.';
+                    }  
+                },
+                {
+                    key: 'time',
+                    prompt: 'How long should the poll last in minutes?',
+                    type: 'integer',
+                    default: 0,
+                    validate: time => {
+                        if (time >= 0 && time <= 60) return true;
```
Comment (MMJ744) [inline commands/polls/polls.js]: I don't think a one hour maximum time is enough for most polls. We have users in most time zones and polls we have done in the past have been open for over a week.

File: commands/polls/polls.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                            .then(async function (message) {
+                                var reactionCountsArray = [];
+                                for (var i = 0; i < optionsList.length; i++) {
+                                    reactionCountsArray[i] = message.reactions.get(emojiList[i]).count-1;
+                                }
+
+                                // Find winner(s)
+                                var max = -Infinity, indexMax = [];
+                                for(var i = 0; i < reactionCountsArray.length; ++i)
+                                    if(reactionCountsArray[i] > max) max = reactionCountsArray[i], indexMax = [i];
```
Comment (MMJ744) [inline commands/polls/polls.js]: I dont know JavaScript that well but I don't think this is correct syntax, even if it is correct its very unclear. Please use a code block for multiple operations, or potentially always use them with IF statements. We will formalise a style guide at some point.

File: commands/polls/vote.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            memberName: 'vote',
+            description: "Starts a yes/no/don't care vote.",
+            examples: ['!vote "Do you like to vote?" "I mean who doesn\'t right?!" 5'],
+            args: [
+                {
+                    key: 'question',
+                    prompt: 'What is the vote question?',
+                    type: 'string',
+                    validate: question => {
+                        if (question.length < 101 && question.length > 11) return true;
```
Comment (MMJ744) [inline commands/polls/vote.js]: Same as with the other poll, I see no reason to place a minimum size on the question.

File: commands/polls/polls.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                    type: 'string',
+                    validate: options => {
+                        var optionsList = options.split(",");
+                        if (optionsList.length > 1) return true;
+                        return 'Polling options must be greater than one.';
+                    }  
+                },
+                {
+                    key: 'time',
+                    prompt: 'How long should the poll last in minutes?',
```
Comment (MMJ744) [inline commands/polls/polls.js]: Should this not also be marked optional like in vote?

File: commands/polls/vote.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                        return 'Polling questions must be between 10 and 200 characters in length.';
+                    }
+                },
+                {
+                    key: 'time',
+                    prompt: '(Optional) How long should the vote last in minutes?',
+                    type: 'integer',
+                    default: 0,
+                    validate: time => {
+                        if (time >= 0 && time <= 60) return true;
```
Comment (MMJ744) [inline commands/polls/vote.js]: Again 1 hour is likely not long enough.

File: commands/polls/vote.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                    setTimeout(() => {
+                        // Re-fetch the message and get reaction counts
+                        message.channel.fetchMessage(message.id)
+                            .then(async function (message) {
+                                var reactionCountsArray = [];                               
+                                for (var i = 0; i < reactionArray.length; i++) {
+                                    reactionCountsArray[i] = message.reactions.get(emojiList[i]).count-1;
+                                }
+                                
+                                // Find winner(s)
```
Comment (MMJ744) [inline commands/polls/vote.js:81]: Much of this code is the same as in poll.js, could this not be reused some how?

**Review by MMJ744 — COMMENTED** [review]
I have had a quick look through, will look more in-depth at some point tomorrow.

File: commands/polls/polls.js (hunk 1 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: k ill change it tmrw

File: commands/polls/polls.js (hunk 2 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: k i will change it

File: commands/polls/polls.js (hunk 3 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: ok ill see what i can do

File: commands/polls/polls.js (hunk 5 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: it should yeah

File: commands/polls/vote.js (hunk 7 above)
Comment (SizzlinSeal01) [reply commands/polls/vote.js:81]: I will look into it, but i dont think you can really. I am also pretty new to node.js

Comment (TheBusyBiscuit) [reply commands/polls/vote.js:81]: Matt is right here, lots of this can and should be refactored.

File: commands/polls/vote.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                                var max = -Infinity, indexMax = [];
+                                for(var i = 0; i < reactionCountsArray.length; ++i)
+                                    if(reactionCountsArray[i] > max) max = reactionCountsArray[i], indexMax = [i];
+                                    else if(reactionCountsArray[i] === max) indexMax.push(i);
```
Comment (TheBusyBiscuit) [inline commands/polls/vote.js]: This is really not a good way to structure code.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Matt already pointed out most of the stuff, but yeah this code is pretty poorly structured and should probably be refactored before merging.
Otherwise it would become a pain to maintain.

File: commands/polls/polls.js (hunk 1 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: fixed

File: commands/polls/vote.js (hunk 4 above)
Comment (SizzlinSeal01) [reply commands/polls/vote.js]: fixed

File: commands/polls/polls.js (hunk 2 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: fixed

File: commands/polls/vote.js (hunk 6 above)
Comment (SizzlinSeal01) [reply commands/polls/vote.js]: fixed

File: commands/polls/polls.js (hunk 3 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: fixed in vote.js, fixing it in polls.js rn

Comment (SizzlinSeal01) [reply commands/polls/polls.js]: done

File: commands/polls/vote.js (hunk 8 above)
Comment (SizzlinSeal01) [reply commands/polls/vote.js]: fixed

File: commands/polls/polls.js (hunk 5 above)
Comment (SizzlinSeal01) [reply commands/polls/polls.js]: fixed

**Comment by brettsaunders21** [comment]
Closed due to Invalid and Inactive

**Pull request closed without merging**